### PR TITLE
feat(cheatcode) Add cheatcode to set block.blobbasefee

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
related with https://github.com/foundry-rs/foundry/issues/7002

is there a way to automatically generate what should be in thecrates/cheatcodes/assets/cheatcodes.json or i have to write ths myself?

### CODE REVIEW

File: crates/cheatcodes/src/evm.rs [hunk 1]
```diff
@@ -298,7 +301,23 @@ impl Cheatcode for getBlockTimestampCall {
...
+        let Self { newBlobBaseFee } = self;
+
+        ensure!(
+            ccx.data.env.cfg.spec_id >= SpecId::CANCUN,
+            "`blobBaseFee` is not supported before the Cancun hard fork; \
+             see relevant EIP for blob base fee"
+        );
 
+        let excess_blob_gas = (*newBlobBaseFee).to_ethers();
+        let c: u64 = excess_blob_gas.to_string().parse().unwrap();
```
Comment (mattsse) [inline crates/cheatcodes/src/evm.rs:315]: this is not necessary and can be replaced with a `.to()` call

**Review by mattsse — CHANGES_REQUESTED** [review]
vm.sol updates are generated

we also want a vm.getBlobBasefee

File: crates/cheatcodes/src/evm.rs (hunk 1 above)
Comment (DoTheBestToGetTheBest) [reply crates/cheatcodes/src/evm.rs:315]: > this is not necessary and can be replaced with a `.to()` call

oh yeah correct thanks for this

**Comment by DoTheBestToGetTheBest** [comment]
> vm.sol updates are generated

it's not generated on my file, should i run a command to generate it or something?

**Comment by DoTheBestToGetTheBest** [comment]
> we also want a vm.getBlobBasefee

this should be on this pr or seperate pr ?

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse does it look better to you ?

**Review by onbjerg — CHANGES_REQUESTED** [review]
looks good! as @mattsse mentioned, we need a `getBlobBaseFee` too, since solc sometimes optimizes away certain reads on `block`, so we (unfortunately) have to add getter cheatcodes too.

**Comment by kamuik16** [comment]
> @yash-atreya could you please bring this over the line, I think this is almost done

hey! I got it here https://github.com/foundry-rs/foundry/pull/7598!

**Comment by onbjerg** [comment]
Closing since this has already been merged per above

**Pull request closed without merging**